### ISSUE TITLE
When inserting line error happens.

### ISSUE THREAD

**Issue by kant2002** [Microsoft/nodejstools#294]
I run latest debug build.

After inserting following line: (from clipboard)
```
: function(scope, element, attrs, controllers) {
```
in the code block below
```
        templateUrl: '/app/modules/food-distribution/card-input.html',
        <<<INSERT HERE>>>
        controller: function ($scope, $element, $attrs) {
```

following error happens:
```
Before looping through characters, we have specified the end point before the start point.  We can't look at a negative range to find the index of newlines.
   at Microsoft.NodejsTools.Formatting.FormattingVisitor.FixStatementIndentation(Int32 prevStart, Int32 end)
   at Microsoft.NodejsTools.Formatting.FormattingVisitor.WalkStatements(Int32 startIndex, IEnumerable`1 stmts, Boolean isMultiLine)
   at Microsoft.NodejsTools.Formatting.FormattingVisitor.WalkStatements(Node node, IEnumerable`1 stmts, Boolean isMultiLine)
   at Microsoft.NodejsTools.Formatting.FormattingVisitor.Walk(ObjectLiteral node)
   at Microsoft.NodejsTools.Parsing.ObjectLiteral.Walk(AstVisitor visitor)
   at Microsoft.NodejsTools.Formatting.FormattingVisitor.Walk(ReturnNode node)
   at Microsoft.NodejsTools.Parsing.ReturnNode.Walk(AstVisitor visitor)
   at Microsoft.NodejsTools.Formatting.FormattingVisitor.WalkStatements(Int32 startIndex, IEnumerable`1 stmts, Boolean isMultiLine)
   at Microsoft.NodejsTools.Formatting.FormattingVisitor.WalkStatements(Node node, IEnumerable`1 stmts, Boolean isMultiLine)
   at Microsoft.NodejsTools.Formatting.FormattingVisitor.WalkBlock(Block block)
   at Microsoft.NodejsTools.Formatting.FormattingVisitor.Walk(FunctionObject node)
   at Microsoft.NodejsTools.Parsing.FunctionObject.Walk(AstVisitor walker)
   at Microsoft.NodejsTools.Formatting.FormattingVisitor.WalkStatements(Int32 startIndex, IEnumerable`1 stmts, Boolean isMultiLine)
   at Microsoft.NodejsTools.Formatting.FormattingVisitor.WalkStatements(Node node, IEnumerable`1 stmts, Boolean isMultiLine)
   at Microsoft.NodejsTools.Formatting.FormattingVisitor.Format(JsAst ast)
   at Microsoft.NodejsTools.Formatting.Formatter.GetEdits(String code, FormattingOptions options, JsAst ast, Boolean onEnter)
   at Microsoft.NodejsTools.Formatting.Formatter.GetEdits(String code, FormattingOptions options, Boolean onEnter)
   at Microsoft.NodejsTools.Formatting.Formatter.GetEditsForRange(String code, Int32 start, Int32 end, FormattingOptions options)
   at Microsoft.NodejsTools.EditFilter.FormatAfterPaste(ITextSnapshot curVersion)
   at Microsoft.NodejsTools.EditFilter.Paste(Guid& pguidCmdGroup, UInt32 nCmdID, UInt32 nCmdexecopt, IntPtr pvaIn, IntPtr pvaOut, Int32& hr)
   at Microsoft.NodejsTools.EditFilter.Exec(Guid& pguidCmdGroup, UInt32 nCmdID, UInt32 nCmdexecopt, IntPtr pvaIn, IntPtr pvaOut)
   at Microsoft.VisualStudio.Editor.Implementation.CommandChainNode.InnerExec(Guid& pguidCmdGroup, UInt32 nCmdID, UInt32 nCmdexecopt, IntPtr pvaIn, IntPtr pvaOut)
   at Microsoft.VisualStudio.Editor.Implementation.SimpleTextViewWindow.Exec(Guid& pguidCmdGroup, UInt32 nCmdID, UInt32 nCmdexecopt, IntPtr pvaIn, IntPtr pvaOut)
   at Microsoft.VisualStudio.Editor.Implementation.CompoundTextViewWindow.Exec(Guid& pguidCmdGroup, UInt32 nCmdID, UInt32 nCmdexecopt, IntPtr pvaIn, IntPtr pvaOut)
   at Microsoft.VisualStudio.Platform.WindowManagement.DocumentObjectSite.Exec(Guid& pguidCmdGroup, UInt32 nCmdID, UInt32 nCmdexecopt, IntPtr pvaIn, IntPtr pvaOut)
   at Microsoft.VisualStudio.Platform.WindowManagement.WindowFrame.Exec(Guid& pguidCmdGroup, UInt32 nCmdID, UInt32 nCmdexecopt, IntPtr pvaIn, IntPtr pvaOut)
```

and subsequentally:
```
---------------------------
Assertion Failed: Abort=Quit, Retry=Debug, Ignore=Continue
---------------------------
edits should be provided in order
   at Microsoft.NodejsTools.Formatting.FormattingVisitor.MaybeReplaceText(Int32 start, Int32 end, String newText)
   at Microsoft.NodejsTools.Formatting.FormattingVisitor.ReplaceFollowingWhiteSpace(Int32 startIndex, String whiteSpace)
   at Microsoft.NodejsTools.Formatting.FormattingVisitor.FixStatementIndentation(Int32 prevStart, Int32 end)
   at Microsoft.NodejsTools.Formatting.FormattingVisitor.WalkStatements(Int32 startIndex, IEnumerable`1 stmts, Boolean isMultiLine)
   at Microsoft.NodejsTools.Formatting.FormattingVisitor.WalkStatements(Node node, IEnumerable`1 stmts, Boolean isMultiLine)
   at Microsoft.NodejsTools.Formatting.FormattingVisitor.WalkBlock(Block block)
   at Microsoft.NodejsTools.Formatting.FormattingVisitor.Walk(FunctionObject node)
   at Microsoft.NodejsTools.Parsing.FunctionObject.Walk(AstVisitor walker)
   at Microsoft.NodejsTools.Formatting.FormattingVisitor.WalkStatements(Int32 startIndex, IEnumerable`1 stmts, Boolean isMultiLine)
   at Microsoft.NodejsTools.Formatting.FormattingVisitor.WalkStatements(Node node, IEnumerable`1 stmts, Boolean isMultiLine)
   at Microsoft.NodejsTools.Formatting.FormattingVisitor.Format(JsAst ast)
   at Microsoft.NodejsTools.Formatting.Formatter.GetEdits(String code, FormattingOptions options, JsAst ast, Boolean onEnter)
   at Microsoft.NodejsTools.Formatting.Formatter.GetEdits(String code, FormattingOptions options, Boolean onEnter)
   at Microsoft.NodejsTools.Formatting.Formatter.GetEditsForRange(String code, Int32 start, Int32 end, FormattingOptions options)
   at Microsoft.NodejsTools.EditFilter.FormatAfterPaste(ITextSnapshot curVersion)
   at Microsoft.NodejsTools.EditFilter.Paste(Guid& pguidCmdGroup, UInt32 nCmdID, UInt32 nCmdexecopt, IntPtr pvaIn, IntPtr pvaOut, Int32& hr)
   at Microsoft.NodejsTools.EditFilter.Exec(Guid& pguidCmdGroup, UInt32 nCmdID, UInt32 nCmdexecopt, IntPtr pvaIn, IntPtr pvaOut)
   at Microsoft.VisualStudio.Editor.Implementation.CommandChainNode.InnerExec(Guid& pguidCmdGroup, UInt32 nCmdID, UInt32 nCmdexecopt, IntPtr pvaIn, IntPtr pvaOut)
   at Microsoft.VisualStudio.Editor.Implementation.SimpleTextViewWindow.Exec(Guid& pguidCmdGroup, UInt32 nCmdID, UInt32 nCmdexecopt, IntPtr pvaIn, IntPtr pvaOut)
   at Microsoft.VisualStudio.Editor.Implementation.CompoundTextViewWindow.Exec(Guid& pguidCmdGroup, UInt32 nCmdID, UInt32 nCmdexecopt, IntPtr pvaIn, IntPtr pvaOut)
   at Microsoft.VisualStudio.Platform.WindowManagement.DocumentObjectSite.Exec(Guid& pguidCmdGroup, UInt32 nCmdID, UInt32 nCmdexecopt, IntPtr pvaIn, IntPtr pvaOut)
   at Microsoft.VisualStudio.Platform.WindowManagement.WindowFrame.Exec(Guid& pguidCmdGroup, UInt32 nCmdID, UInt32 nCmdexecopt, IntPtr pvaIn, IntPtr pvaOut)
```

**Comment by kant2002** [comment]
@mousetraps please advice how to start debugging such issues

**Comment by mjbvz** [comment]
@kant2002 Can you please try to repo this issue with NTVS 1.2.

I just tried to repo it again using the code:

```
var a = {
    templateUrl: '/app/modules/food-distribution/card-input.html',

    controller: function ($scope, $element, $attrs) {}
};
````

and inserting ` : function (scope, element, attrs, controllers) {` in to the space between the properties. I could not get this error with either ES6 preview Intellisense or with Full Intellisense, but did discover another fun bug if you just insert the text `function (scope, element, attrs, controllers) {`

**Comment by mjbvz** [comment]
I believe this is fixed in NTVS 1.2 Beta with ES6 IntelliSense. I was unable to repo the original problem or my slightly modified example case.
